drivers/mtd/gd25.c

```diff
@@ -107,6 +107,7 @@
 #define GD25_SR_WIP                 (1 << 0)  /* Bit 0: Write in Progress */
 #define GD25_SR_WEL                 (1 << 1)  /* Bit 1: Write Enable Latch */
 #define GD25_SR1_EN4B               (1 << 3)  /* Bit 3: Enable 4byte address */
+#define GD25Q_SR1_EN4B              (1 << 0)  /* Bit 0: Enable 4byte address GD25Q memories */
```

Comment:
It is not correct for all GD25Q devices GD25Q32 has WIP as bit 0: https://www.elm-tech.com/en/products/spi-flash-memory/gd25q32/gd25q32.pdf

Reply:
This isn't about the WIP bit, it's the 4 byte read bit, which is only relevant to memories with >128mbit.

It's in the second status register, which has different names dependent on the device, and either bit 0 or 3.

The driver will only attempt to read or write this if the size is >128Mbit

Reply:
Yes, I know it is not WIP, what I said is: in GD25G32 the bit 0 is not EN4B, but WIP:
![gd25](https://user-images.githubusercontent.com/37000/203980016-be3ff6bc-ae6e-4236-85cf-139023cd9eb1.png)

Reply:
So, you cannot assume that all GD25Q devices will have EN4B at bit 0 position.

Reply:
Apologies, perhaps what you wrote was ambiguous and I misunderstood - but that is not the bit that is read (it reads the second status reg to get S11 - or S8 with this change) BUT it will not be read for a 32Mbit device anyway. The driver will never attempt to place this device into 4 byte read mode, so it is not a problem:

      if (priv->nsectors > GD25_NSECTORS_128MBIT)
        {          
           if (!gd25_4ben(priv))
             {
               ferr("ERROR: capacity %02x: Can't enable 4-byte mode!\n", capacity);
               ... etc

Reply:
Thank you @TimJTi understood the issue!